src/components/data-management/AddMetadataButton.jsx

```diff
@@ -17,31 +21,52 @@ const AddMetadataButton = ({ samplesTableRef }) => {
   const isSubsetted = activeExperiment?.isSubsetted;
   const samples = useSelector((state) => state.samples);
   const selectedTech = samples[activeExperiment?.sampleIds[0]]?.type;
-  const [uploadModalVisible, setUploadModalVisible] = useState(false);
 
-  const uploadFiles = (file) => {
+  const [uploadModalVisible, setMetadataUploadModalVisible] = useState(false);
+  const [cellLevelUploadVisible, setCellLevelUploadVisible] = useState(false);
+
+  const onUploadMetadataTrack = (file) => {
     dispatch(uploadMetadataFile(activeExperimentId, file));
```

Comment:
What do you think about renaming this action to `uploadMetadataTrackFile` or `uploadMetadataTrack`?

Reply:
i prefer keeping it like this. The word 'on' suggests its a handler function